site/zenodo_rdm/support/email.py

```diff
@@ -0,0 +1,100 @@
+from flask import current_app, request
+from flask_login import current_user
+from flask_mail import Message
+from invenio_accounts.sessions import _extract_info_from_useragent
+
+from .utils import format_user_email, render_template_to_string
+
+
+class EmailService(object):
+    """Basic service to send an e-mail using the configured mail extension."""
+    @property
+    def mail(self):
+        service = current_app.extensions["mail"]
+        if not service:
+            raise Exception("No mail extension provided.")
+        return service
+
+    def send_email(
+        self, title, recipients, sender, body, reply_to="", attachments=None
+    ):
+        """Sends an e-mail to the given recipient(s)."""
+        if type(recipients) != list:
+            recipients = [recipients]
+
+        msg = Message(
+            title, sender=sender, recipients=recipients, reply_to=reply_to, body=body
+        )
+        if attachments:
+            for upload in attachments:
+                msg.attach(upload.filename, "application/octet-stream", upload.read())
+        self.mail.send(msg)
+
+
+class SupportEmailService(EmailService):
+    """_summary_
+    Provides an interface to send support related emails.
+    """
+    def __init__(self):
+        """Constructor."""
+        super().__init__()
+
+    def send_support_email(
```

Comment:
Perhaps this can be simplified into a method that works solely with contexts, as suggested by @slint personally.